storage/src/vespa/storage/frameworkimpl/thread/deadlockdetector.h

```diff
@@ -81,8 +79,8 @@ struct DeadLockDetector : private framework::Runnable,
     std::condition_variable _cond;
     bool _enableWarning;
     bool _enableShutdown;
-    std::atomic<uint64_t> _processSlackMs;
-    std::atomic<uint64_t> _waitSlackMs;
+    std::atomic<vespalib::duration> _processSlackMs;
+    std::atomic<vespalib::duration> _waitSlackMs;
```

Comment:
Test added, they are lock free. However I do agree on the deadlock detector. It is overly complicated. It will also have less value when we go fully async, as different threads will experience the trouble.
We might want to consider its right to live later on.